catalog/to-k8s/source_test.go

```diff
@@ -98,20 +115,29 @@ func TestSource_ignoreK8S(t *testing.T) {
 	t.Parallel()
 	require := require.New(t)
 
-	a := agent.NewTestAgent(t, t.Name(), ``)
-	defer a.Shutdown()
-	testrpc.WaitForTestAgent(t, a.RPC, "dc1")
-	client := a.Client()
+	// Set up server, client
+	a, err := testutil.NewTestServerT(t)
+	require.NoError(err)
+	defer a.Stop()
+
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPAddr,
+	})
+	require.NoError(err)
 
 	// Create services before the source is running
-	_, err := client.Catalog().Register(testRegistration("hostA", "svcA", nil), nil)
+	_, err = client.Catalog().Register(testRegistration("hostA", "svcA", nil), nil)
 	require.NoError(err)
 	_, err = client.Catalog().Register(testRegistration("hostB", "svcA", nil), nil)
 	require.NoError(err)
 	_, err = client.Catalog().Register(testRegistration("hostB", "svcB", []string{toconsul.TestConsulK8STag}), nil)
 	require.NoError(err)
 
-	_, sink, closer := testSource(t, client)
+	// Setting a different domain for each test so that
+	// this test is safe to run in parallel
+	_, sink, closer := testSourceWithConfig(client, func(s *Source) {
+		s.Domain = "ignoreK8S"
+	})
 	defer closer()
 
 	var actual map[string]string
```

Comment:
Just below here `len(actual) == 0` needs to be `len(actual) != 2` so that it waits until all services are synced.

Reply:
Updated.